tools/test/toolchains/test_toolchains.py

```diff
@@ -40,6 +41,58 @@ def test_arm_minimal_printf(self):
         self.assertIn("-DMBED_MINIMAL_PRINTF", arm_micro_obj.flags["common"])
         self.assertIn("-DMBED_MINIMAL_PRINTF", arm_c6_obj.flags["common"])
 
+    def test_arm_default_lib(self):
+        """Test that linker flags are correctly added to an instance of ARM."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.supported_c_libs = {"arm": ["small"]}
+        mock_target.default_lib = "small"
+        mock_target.default_toolchain = "ARM"
+        mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5", "ARMC6"]
+        arm_std_obj = ARM_STD(mock_target)
+        arm_micro_obj = ARM_MICRO(mock_target)
+
+        mock_target.default_toolchain = "ARMC6"
+        arm_c6_obj = ARMC6(mock_target)
+
+        self.assertIn("-D__MICROLIB", arm_std_obj.flags["common"])
+        self.assertIn("-D__MICROLIB", arm_micro_obj.flags["common"])
+        self.assertIn("-D__MICROLIB", arm_c6_obj.flags["common"])
+
+        self.assertIn("--library_type=microlib", arm_std_obj.flags["ld"])
+        self.assertIn("--library_type=microlib", arm_micro_obj.flags["ld"])
+        self.assertIn("--library_type=microlib", arm_c6_obj.flags["ld"])
+
+        self.assertIn("-Wl,--library_type=microlib", arm_c6_obj.flags["cxx"])
+        self.assertIn("--library_type=microlib", arm_c6_obj.flags["asm"])
+
+    def test_arm_default_lib_std_exception(self):
+        """Test that exception raised when default_lib is std but supported_c_libs parameter arm is not suppoted std lib."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5"]
+        mock_target.default_toolchain = "ARM"
+        mock_target.default_lib = "std"
+        mock_target.supported_c_libs = {"arm": ["small"]}
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_DEFAULT_LIB_EXECPTION_STRING.format(mock_target.default_lib)):
+            ARM_STD(mock_target)
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_DEFAULT_LIB_EXECPTION_STRING.format(mock_target.default_lib)):
+            ARMC6(mock_target)
+
+
+    def test_arm_default_lib_small_exception(self):
+        """Test that exception raised when default_lib is small but supported_c_libs parameter arm is not suppoted small lib."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.default_lib = "small"
```

Comment:
To also cover case insensitivity
```suggestion
        mock_target.default_lib = "sMAll"
```